src/diffusers/models/unet_2d_condition.py

```diff
@@ -398,6 +417,7 @@ def __init__(
                 resnet_skip_time_act=resnet_skip_time_act,
                 resnet_out_scale_factor=resnet_out_scale_factor,
                 cross_attention_norm=cross_attention_norm,
+                attention_head_dim=attention_head_dim[i] if attention_head_dim[i] is not None else output_channel,
```

Comment:
```suggestion
                attention_head_dim=output_channel //num_attention_heads[i],
```
so not sure what this is intended to do here, I think `attention_head_dim` would never be `None` in our case, so the default `8` would just be passed down as it is and used to calculate the number of attention heads 

anyways this prints out `40`, (320//8), I think if we pass `num_attention_heads = 16` we would want to see the number of attentions to be heads to be 16

```python
down_block_types = ("AttnDownBlock2D",)
up_block_types = ("AttnUpBlock2D",)
unet = UNet2DConditionModel(
    num_attention_heads = 16,
    block_out_channels = (320,),
    down_block_types = down_block_types,
    up_block_types = up_block_types)
# this prints 40
unet.down_blocks[0].attentions[0].heads
```

Reply:
https://github.com/huggingface/diffusers/pull/3797#discussion_r1237056350

Reply:
Note that by default `attention_head_dim` is defined to be `8` and it takes priority for classes such as `AttnDownBlock2D` as mentioned by @pcuenca in #3797 

However this is indeed confusing as now we have some attention blocks where `num_attention_heads` take priority, e.g. the cross attention blocks and some where number of attention heads take priority. I will clean this up in a follow-up PR

Reply:
@yiyixuxu , this:

```py
attention_head_dim=output_channel //num_attention_heads[i],
```

would break current behavior, e.g. configs that have `attention_head_dim` set to None and would then pass an incorrect number here